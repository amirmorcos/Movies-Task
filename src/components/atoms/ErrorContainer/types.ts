import { StyleProp, ViewStyle } from "react-native";

export type ErrorContainerProps = {
  onRefetch: () => void;
  overrideContainerStyle?: StyleProp<ViewStyle>;
};
