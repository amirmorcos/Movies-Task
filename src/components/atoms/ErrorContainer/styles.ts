import { StyleSheet } from "react-native";
import { moderateScale, s } from "react-native-size-matters";

export default StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "black",
    justifyContent: "center",
    alignItems: "center",
  },
  text: {
    color: "white",
    fontSize: moderateScale(15),
    marginEnd: s(5),
  },
  retryContainer: {
    flexDirection: "row",
    alignItems: "center",
  },
});
