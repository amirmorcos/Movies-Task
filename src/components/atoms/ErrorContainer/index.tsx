import React from "react";
import { Text } from "react-native";
import { TouchableOpacity, View } from "react-native";
import { ErrorContainerProps } from "./types";
import FontAwesome6 from "react-native-vector-icons/FontAwesome6";
import styles from "./styles";

const ErrorContainer = ({
  onRefetch,
  overrideContainerStyle,
}: ErrorContainerProps) => {
  return (
    <View style={[styles.container, overrideContainerStyle]}>
      <TouchableOpacity
        onPress={() => onRefetch()}
        style={styles.retryContainer}
      >
        <Text style={styles.text}>Try again</Text>
        <FontAwesome6 name="arrow-rotate-right" color="white" />
      </TouchableOpacity>
    </View>
  );
};

export default ErrorContainer;
