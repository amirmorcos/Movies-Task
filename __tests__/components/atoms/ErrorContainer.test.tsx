import ErrorContainer from "atoms/ErrorContainer";
import React from "react";
import renderer from "react-test-renderer";

jest.mock("react-native-vector-icons/FontAwesome6", () => "Icon");
jest.mock("react-native-size-matters", () => ({
  s: jest.fn(),
  vs: jest.fn(),
  moderateScale: jest.fn(),
}));

test("renders correctly", () => {
  const tree = renderer.create(<ErrorContainer onRefetch={jest.fn} />).toJSON();
  expect(tree).toMatchSnapshot();
});
