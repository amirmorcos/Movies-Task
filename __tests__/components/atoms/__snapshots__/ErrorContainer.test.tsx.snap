// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "black",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": undefined,
          "marginEnd": undefined,
        }
      }
    >
      Try again
    </Text>
    <Icon
      color="white"
      name="arrow-rotate-right"
    />
  </View>
</View>
`;
